### PR TITLE
repo.active_stages should return active stages

### LINKED ISSUE #2046: gc: unused arguments
Might be a bug https://github.com/iterative/dvc/pull/2029#issuecomment-494268568

### LINKED ISSUE #2356: dvc: looks like Repo.active_stages() is not working
It calls `.pipelines()`, which uses the whole graph instead of active one. That could be fixed, I would however say that removing distinction between active and all stages for purpose of push/pull/gs/status might be a good idea.

### PR DESCRIPTION
**Pull request by pared**
* [ ] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [ ] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----
Fixes #2356 
Fixes #2046

### CODE REVIEW

File: dvc/repo/__init__.py [hunk 1]
```diff
@@ -353,15 +353,25 @@ def graph(self, stages=None, from_directory=None):
...
+            graph.subgraph(c).copy()
+            for c in nx.weakly_connected_components(graph)
         ]
 
+    def pipelines(self, from_directory=None):
+        G, _ = self.graph(from_directory=from_directory)
+
+        return self._pipelines(G)
+
+    def pipelines_for_active_nodes(self, from_directory=None):
```
Comment (efiop) [inline dvc/repo/__init__.py]:
```suggestion
    def active_pipelines(self, from_directory=None):
```
?

File: dvc/repo/gc.py [hunk 2]
```diff
@@ -21,9 +21,7 @@ def _merge_cache_lists(clists):
 
 def _load_all_used_cache(
     repos,
-    target=None,
```
Comment (efiop) [inline dvc/repo/gc.py]: Fixes #2046

File: dvc/repo/__init__.py [hunk 3]
```diff
@@ -353,15 +353,25 @@ def graph(self, stages=None, from_directory=None):
 
         return G, G_active
 
-    def pipelines(self, from_directory=None):
+    @staticmethod
```
Comment (efiop) [inline dvc/repo/__init__.py]: Or maybe
```
def pipelines(self, from_directory=None, active=False):
    import networkx as nx

    G = self.graph(from_directory=from_directory, active=active)

    return [
        G.subgraph(c).copy() for c in nx.weakly_connected_components(G)
    ]
```

and then use it down below as
```
for G in self.pipelines(from_directory, active=True):
```
?

**Comment by Suor** [comment]
@pared @efiop I suggest dropping active stages notion and keeping current behavior in #2356, the full story is there.

File: dvc/repo/__init__.py (hunk 3 above)
Comment (pared) [reply dvc/repo/__init__.py]: I believe you are right

**Comment by efiop** [comment]
@pared Could you check docs for affected commands to see if they need to be corrected?

**Comment by pared** [comment]
@efiop I checked the docs, it is not mentioned there whether we update dvc state on dvc move. Examples do not contain `status` check. 

There is no need to update the docs.

We can note that move will not affect `status`, though I believe that it is quite intuitive that it should not, and the original case was a bug in `move`.

**Comment by efiop** [comment]
@pared What `dvc state`? I meant that we might mention in the docs that we don't push/pull/etc dependencies of locked stages, which wasn't true because of the bug and now is not true because we've decided so.

File: dvc/repo/__init__.py [hunk 4]
```diff
@@ -225,20 +224,10 @@ def used_cache(
...
                         )
-            elif active:
-                stages = self.active_stages()
             else:
                 stages = self.stages()
 
             for stage in stages:
-                if active and not targets and stage.locked:
-                    logger.warning(
-                        "DVC-file '{path}' is locked. Its dependencies are"
```
[the remark this answers is not in the document]
Comment (pared) [reply dvc/repo/__init__.py]: No, it does not.

**Pull request merged**